src/wp-includes/blocks.php

```diff
@@ -853,22 +853,33 @@ function get_hooked_blocks() {
 }
 
 /**
- * Returns the markup for blocks hooked to the given anchor block in a specific relative position.
+ * Determines whether there are any blocks hooked to other blocks.
+ * Prefixed with maybe_ because the filter might be used but not have any hooked blocks.
  *
- * @since 6.5.0
- * @access private
+ * @since 6.7.0
  *
- * @param array                   $parsed_anchor_block The anchor block, in parsed block array format.
- * @param string                  $relative_position   The relative position of the hooked blocks.
- *                                                     Can be one of 'before', 'after', 'first_child', or 'last_child'.
- * @param array                   $hooked_blocks       An array of hooked block types, grouped by anchor block and relative position.
- * @param WP_Block_Template|array $context             The block template, template part, or pattern that the anchor block belongs to.
- * @return string
+ * @return bool
  */
-function insert_hooked_blocks( &$parsed_anchor_block, $relative_position, $hooked_blocks, $context ) {
-	$anchor_block_type  = $parsed_anchor_block['blockName'];
-	$hooked_block_types = isset( $hooked_blocks[ $anchor_block_type ][ $relative_position ] )
-		? $hooked_blocks[ $anchor_block_type ][ $relative_position ]
+function maybe_has_hooked_blocks() {
+	$hooked_blocks = get_hooked_blocks();
```

Comment:
`get_hooked_blocks()` is now run here and inside `get_hooked_blocks_by_anchor_block` which is executed on every block traversal. I think we need to figure out a way to either avoid this, or a caching strategy around it.

Reply:
In this commit (https://github.com/WordPress/wordpress-develop/pull/6584/commits/6bdec1e59ad3acff107244119a53aea06cb7ba4f) I have memoized the result of `get_hooked_blocks()`. The cache will last the persist per-request which means we won't be degrading performance when comparing it to how it currently works I don't think.